src/App.tsx

```diff
@@ -137,11 +143,15 @@ const Providers: React.FC = ({ children }) => {
                                     <DpiIndexComponentsProvider>
                                       <CgiTokenMarketDataProvider>
                                         <CgiIndexComponentsProvider>
-                                          <IndexTokenMarketDataProvider>
-                                            <SnapshotProposalsProvider>
-                                              {children}
-                                            </SnapshotProposalsProvider>
-                                          </IndexTokenMarketDataProvider>
+                                          <MviTokenMarketDataProvider>
```

Comment:
I'm curious, why is this wrapped inside the other components? Aren't these new components peers, of say `CgiTokenMarketDataProvider`? It may be a little confusing that the structure of the program doesn't match the hierarchy.

Reply:
Yup, all the market data/component providers are peers. Would love to dig in to how this might be better structured. It's not a problem right now but this is also leading to duplicate API calls since fetching happens in the provider